includes/wikia/parser/templatetypes/handlers/InfoiconTemplate.class.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+class InfoiconTemplate {
+	/**
+	 * @desc sanitize infoicon template content, that is remove all non-images
+	 * from it's wikitext. They usually contain span with 'display: none' style
+	 * or other stuff we don't want on mobile.
+	 *
+	 * @param string $wikitext
+	 *
+	 * @return string wikitext with image(s) only
+	 */
+	public static function handle( $wikitext ) {
+		global $wgContLang;
+
+		$images = FileNamespaceSanitizeHelper::getInstance()->getCleanFileMarkersFromWikitext( $wikitext, $wgContLang );
+
+		if ( $images ) {
+			$sizedImages = array_map( function( $img ) {
+				return '[[' . $img . '|30px]]';
```

Comment:
Like we talked about, me no gusta as Mercury does not have any differentiation between infoicon and small image in this case.
I'll try to propose some tweaks tomorrow, for now it looks ok-ish